### PR TITLE
Add the code for running MGD with OpenAI models

### PR DESCRIPTION
**Pull request by LakshyAAAgrawal**

### CODE REVIEW

File: src/monitors4codegen/monitor_guided_decoding/openai_gen.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    assert len(prompt_tokenized.shape) == 1
+
+    all_tokens: torch.Tensor = prompt_tokenized
+    gen_text: bytes = b''
+    
+    gen_tokens: List[int] = []
+
+    tokens_sort_key = {k:[0, 0] for k in tokenizer.all_token_ids}
+    
+    # # TODO: Find a way to prioritize tokens to be blacklisted
```
Comment (adityakanade) [inline src/monitors4codegen/monitor_guided_decoding/openai_gen.py:41]: Please document why this TODO can improve the current implementation.

File: src/monitors4codegen/monitor_guided_decoding/openai_gen.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        dot_found = False
+        for token_bytes in tokens_gen_bytes_:
+            gen_text += token_bytes
+            all_text_bytes += token_bytes
+            tokens_gen_bytes.append(token_bytes)
+            if b'.' in token_bytes:
+                dot_found = True
+                break
+
+        should_manually_add_dot = None
```
Comment (adityakanade) [inline src/monitors4codegen/monitor_guided_decoding/openai_gen.py:130]: Add a comment to explain what is happening here and the reason to manually add a dot.

File: src/monitors4codegen/monitor_guided_decoding/openai_gen.py (hunk 1 above)
Comment (LakshyAAAgrawal) [reply src/monitors4codegen/monitor_guided_decoding/openai_gen.py:41]: Added in the PR: https://github.com/microsoft/monitors4codegen/pull/16

File: src/monitors4codegen/monitor_guided_decoding/openai_gen.py (hunk 2 above)
Comment (LakshyAAAgrawal) [reply src/monitors4codegen/monitor_guided_decoding/openai_gen.py:130]: Added in PR: https://github.com/microsoft/monitors4codegen/pull/16

**Pull request merged**